zerver/lib/rest.py

```diff
@@ -95,79 +84,107 @@ def rest_dispatch(request: HttpRequest, **kwargs: Any) -> HttpResponse:
-        elif "override_api_url_scheme" in view_flags and request.GET.get("api_key") is not None:
-            # This request uses legacy API authentication.  We
-            # unfortunately need that in the React Native mobile apps,
-            # because there's no way to set the Authorization header in
-            # React Native.  See last block for rate limiting notes.
-            target_function = authenticated_uploads_api_view(skip_rate_limiting=True)(
-                target_function
-            )
-        # /json views (web client) validate with a session token (cookie)
-        elif not request.path.startswith("/api") and request.user.is_authenticated:
-            # Authenticated via sessions framework, only CSRF check needed
-            auth_kwargs = {}
-            if "override_api_url_scheme" in view_flags:
-                auth_kwargs["skip_rate_limiting"] = True
-            target_function = csrf_protect(authenticated_json_view(target_function, **auth_kwargs))
-
-        # most clients (mobile, bots, etc) use HTTP basic auth and REST calls, where instead of
-        # username:password, we use email:apiKey
-        elif "Authorization" in request.headers:
-            # Wrap function with decorator to authenticate the user before
-            # proceeding
-            target_function = authenticated_rest_api_view(
-                allow_webhook_access="allow_incoming_webhooks" in view_flags,
-            )(target_function)
-        elif (
-            request.path.startswith(("/json", "/avatar", "/user_uploads", "/thumbnail"))
-            and "allow_anonymous_user_web" in view_flags
-        ):
-            # For endpoints that support anonymous web access, we do that.
-            # TODO: Allow /api calls when this is stable enough.
-            target_function = csrf_protect(public_json_view(target_function))
-        else:
-            # Otherwise, throw an authentication error; our middleware
-            # will generate the appropriate HTTP response.
-            raise MissingAuthenticationError()
-
-        if request.method in ["DELETE", "PATCH", "PUT"]:
-            # process_as_post needs to be the outer decorator, because
-            # otherwise we might access and thus cache a value for
-            # request.POST.
-            target_function = process_as_post(target_function)
-
-        return target_function(request, **kwargs)
+            return entry
+        return supported_methods[method_to_use], set()
 
     return json_method_not_allowed(list(supported_methods.keys()))
 
 
+@default_never_cache_responses
+@csrf_exempt
+def rest_dispatch(request: HttpRequest, **kwargs: Any) -> HttpResponse:
+    """Dispatch to a REST API endpoint.
+
+    Unauthenticated endpoints should not use this, as authentication is verified
+    in the following ways:
+        * for paths beginning with /api, HTTP basic auth
+        * for paths beginning with /json (used by the web client), the session token
```

Comment:
I added to this commit an update to these comments to no longer inaccurately claim that unauthenticated endpoints can't use it, instead documenting `allow_anonymous_user_web`.